### DISCUSSION TITLE
Error on request data with relationship with doctrine

### DISCUSSION THREAD

**Discussion by GabrielBN123** [symfony/symfony#43504] [Q&A] [answered]
I trying to request a data from db, the basic data I achieved, but the other data I'm having difficulty in bringing, it pulls the id, but the data for this relationship is null. 

My code
![image](https://user-images.githubusercontent.com/54750556/137313736-efc923fe-5dcb-4df2-9d99-b33239bae0cb.png)

the return
![image](https://user-images.githubusercontent.com/54750556/137313935-e45fddf3-b907-4163-a432-fc4b9a0238f3.png)

**Comment by mbabker** [comment, marked as answer]
You have a proxy object on the relationship right now.

Internally, Doctrine uses proxies as a way to lazily load an object’s data on the first read.  So, when you start pulling data from that cliente relation, it will be changed from a proxy object (note the namespace) to the real thing and the data visible in your dump.

**Reply by GabrielBN123** [reply]
sorry, I don't understand what I have to do, I'm new whit symfony

**Reply by ghost** [reply]
@GabrielBN123 
As long as you don't explicitly get the data from `Cliente`:  `$agendaMentos->getClient()`, `Client` properties will be null. This behavior allows you to avoid retrieve data unnecessarily, I think the `id` is always set so that a query can be performed if necessary.
But if you use `Cliente`, the proxy takes care of retrieving the data.

This is the opportunity for you to look at problem n+1 by the way. Search on google [Doctrine n+1](https://www.google.com/search?channel=fs&client=ubuntu&q=Doctrine+n%2B1).

**Reply by GabrielBN123** [reply]
I got it, using fetch="EAGER" 

![image](https://user-images.githubusercontent.com/54750556/137388300-2538c373-7d50-434b-9b03-194c3afcfc8b.png)
